Good-Bids/goodbids
# [#295] Nonprofit Site Verification

# Summary

This PR moves the extra custom Nonprofit Site fields from the Add/Edit Site screen to a new tab on the Edit Site screen. It displays as "Verification" when the site is unverified, and "Details" when the site is Verified. Super Admins will see a toggle allowing them to "verify" a site. Anyone with `manage_sites` access can Add/Edit these details. Once the site is verified, only a Super Admin can modify the fields, otherwise they are disabled. When a site has not been verified, the Admin for that site is not accessible, except only by Super Admins.

## Issues

* #295

## Testing Instructions

* All sites are considered unverified by default, except for the GoodBids Main site.
* Testing should be done as Super Admin and BDP Admin

1. Attempt to access WP Admin for a Nonprofit using a non-Super Admin user
2. Since the site should be disabled by default, you should get a `wp_die()` message.
3. Visit the Network Admin as a BDP Admin user
4. Click Verification for a Nonprofit Site
5. Fill in all the required fields.
6. Make sure it saves and gives you a success notice.
7. As a Super Admin, visit the same page
8. Switch the toggle On next to Verified
9. Save changes and the Nonprofit WP Admin should now be accessible.
10. You can quickly see which sites have been verified and which have not from the Network Admin Sites page.

## Screenshots

<img width="936" alt="Screenshot 2024-02-27 at 10 43 23 AM" src="https://github.com/Good-Bids/goodbids/assets/122309362/67a21e5d-6506-4906-8c36-5b6d843e4aa9">

<img width="649" alt="Screenshot 2024-02-27 at 12 15 32 PM" src="https://github.com/Good-Bids/goodbids/assets/122309362/418aad21-ba96-41cd-a929-d9666ea9027e">

<img width="663" alt="Screenshot 2024-02-27 at 1 26 12 PM" src="https://github.com/Good-Bids/goodbids/assets/122309362/26b73036-197c-4008-8469-8600ac2df94d">

<img width="1094" alt="Screenshot 2024-02-27 at 1 39 31 PM" src="https://github.com/Good-Bids/goodbids/assets/122309362/c49d04c9-e315-442a-8c7e-c21407ec3070">

CCing everyone on this in case you find WP Admin inaccessible after this is merged.
